### PR TITLE
Babel preset configurable

### PR DESCRIPTION
**Pull request by sergiodxa**
I'm not sure if this is desired, but I made the `next/babel` preset configurable to allow changing the `babel-preset-env` targets. This can allow us to only target the last browsers and Node.js versions in development, and still works as usual if you don't configure anything. I also included a really simple example using `const` and only transforming it to `var` in production (`next build`).

> This can improve the development build time.

With this change you can set the preset as usual or this way:
```json
["next/babel", {
  "targets": {
    "browsers": "last 1 Chrome version",
    "node": true
  }
}]
```

### CODE REVIEW

**Comment by HaNdTriX** [comment]
@sergiodxa Thanks a lot!

@arunoda can you take a look at this. Since https://github.com/zeit/next.js/pull/1701 there is no way to configure browser support using [babel-preset-env](https://github.com/babel/babel-preset-env).

**Comment by arunoda** [comment]
@sergiodxa I like to have something like this.
But is it possible to use next-babel as a preset and babel-preset-env as a plugin.

Since we extract pages from the browser build for the server, we may need to do further transformations for the server as well separately.

File: server/build/babel/preset.js [hunk 1]
```diff
@@ -11,10 +11,15 @@ const envPlugins = {
...
-module.exports = {
+const defaultOpts = {
+  targets: {}
+}
+
+module.exports = (context, opts = defaultOpts) => ({
   presets: [
     [require.resolve('babel-preset-env'), {
-      modules: false
+      env: { modules: false },
```
Comment (benhjames) [inline server/build/babel/preset.js]: Should this just be `modules: false`? :o

**Comment by HaNdTriX** [comment]
> But is it possible to use next-babel as a preset and **babel-preset-env as a plugin.**

@arunoda do you have an example how this setup could look like?

**Comment by pronevich** [comment]
@arunoda any plans merge it v3 ?

**Comment by benhjames** [comment]
What happened to this? 😮

**Pull request closed without merging**